plus/provider/src/main/java/org/sakaiproject/plus/ProviderServlet.java

```diff
@@ -0,0 +1,1756 @@
+
+import org.tsugi.deeplink.objects.DeepLinkResponse;
+import org.tsugi.deeplink.objects.LtiResourceLink;
+
+import org.sakaiproject.plus.api.PlusService;
+
+import org.sakaiproject.plus.api.model.Tenant;
+import org.sakaiproject.plus.api.model.Subject;
+import org.sakaiproject.plus.api.model.Context;
+import org.sakaiproject.plus.api.model.Link;
+import org.sakaiproject.plus.api.model.LineItem;
+import org.sakaiproject.plus.api.model.Score;
+
+import org.sakaiproject.plus.api.repository.TenantRepository;
+import org.sakaiproject.plus.api.repository.ContextRepository;
+
+import org.sakaiproject.plus.api.Launch;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+
+@SuppressWarnings("deprecation")
+@Slf4j
+public class ProviderServlet extends HttpServlet {
+
+	private static final long serialVersionUID = 1L;
+	private static ResourceLoader rb = new ResourceLoader("plus");
+	private static final String BASICLTI_RESOURCE_LINK = "blti:resource_link_id";
+
+	private static final String SAKAI_SITE_LAUNCH = "sakai.site";
+	private static final String SAKAI_DEEPLINK_LAUNCH = "sakai.deeplink";
+
+	private static final String DEFAULT_PRIVACY_URL = "https://www.sakailms.com/plus-privacylaunch";
+
+	private static final String CACHE_NAME = ProviderServlet.class.getName() + "_cache";
+
+	// Wait five or 30 minutes between successive calls to NRPS.
+	public final long delayNRPSInstructor = 300;
+	public final long delayNRPSLearner = 30*60;  // 30 minutes
+
+	@Autowired private ServerConfigurationService serverConfigurationService;
+	@Autowired private SiteMembershipUpdater siteMembershipUpdater;
+	@Autowired private SiteMembershipsSynchroniser siteMembershipsSynchroniser;
+	@Autowired private SiteEmailPreferenceSetter siteEmailPreferenceSetter;
+	@Autowired private UserFinderOrCreator userFinderOrCreator;
+	@Autowired private UserLocaleSetter userLocaleSetter;
+	@Autowired private UserPictureSetter userPictureSetter;
+	@Autowired private LTIService ltiService;
+	@Autowired private PlusService plusService;
+	@Autowired private TenantRepository tenantRepository;
+	@Autowired private ContextRepository contextRepository;
+
+	private List<BLTIProcessor> bltiProcessors = new ArrayList();
+
+	private String randomUUID = UUID.randomUUID().toString();
+
+	private KeyPair localKeyPair = LTI13Util.generateKeyPair();
+
+	private static Cache getCache()
+	{
+		CacheManager cacheManager = (CacheManager) ComponentManager.get("org.sakaiproject.ignite.SakaiCacheManager");
+		return cacheManager.getCache(CACHE_NAME);
+	}
+
+	public static String getCacheKey(String key)
+	{
+		Cache cache = getCache();
+		Cache.ValueWrapper value = cache.get(key);
+		if ( value == null ) return null;
+		return (String) value.get();
+	}
+
+	public static void putCacheKey(String key, String value)
+	{
+		Cache cache = getCache();
+		cache.put(key, value);
+	}
+
```

Comment:
These likely don't need to be static

Reply:
I will go one better - they are not even needed at all.  Deleted.